examples/qml_features/tests/tst_custom_base_class.qml

```diff
@@ -82,4 +82,26 @@ TestCase {
         compare(instantiator.count, 1);
         compare(dataChangedSpy.count, 1);
     }
+
+    function test_roles_qenum() {
+        compare(CustomBaseClass.Id, 0);
+        compare(CustomBaseClass.Value, 1);
+
+        const model = createTemporaryObject(componentCustomBaseClass, null, {});
+
+        model.add();
+        compare(model.data(model.index(0, 0), CustomBaseClass.Id), 0);
+        compare(model.data(model.index(0, 0), CustomBaseClass.Value), 0);
+
+        model.add();
+        compare(model.data(model.index(1, 0), CustomBaseClass.Id), 1);
+        compare(model.data(model.index(1, 0), CustomBaseClass.Value), 1.0 / 3.0);
+
+        compare(model.state, CustomBaseClass.Idle);
+        model.addOnThread(1);
+        // This shouldn't cause a flaky test, as the `addOnThread` cannot reset the
+        // state to `Idle` until the Event-Loop of this thread is available again.
+        // Which should only be after this thread has finished.
+        compare(model.state, CustomBaseClass.Running);
```

Comment:
can you add a further line to the end to check that we eventually go back to idle?

```
tryCompare(model.state, CustomBaseClass.Idle);
```

Reply:
Done.
I added a method to configure the delay, so we can use 0 ms in the test to not artificially inflate test runtimes.
Because the qt_thread queues something for the main thread event loop, the check to Running should still work.